docs/e2e_testing.md

```diff
@@ -0,0 +1,44 @@
+# End-to-End Testing Configuration System
+
+This document explains how the end-to-end testing system for VERL has been reorganized to support multiple jobs with shared configurations.
+
+## Configuration Structure
+
+The configuration system uses a hierarchical approach:
+
+1. **Base Configuration** (`gsm8k_base.yaml`): Contains common parameters shared across all GSM8K tests.
+2. **Job-Specific Configurations** (e.g., `gsm8k_function_rm.yaml`): Extend the base configuration with job-specific settings.
+
+## Running Tests
+
+Tests can be run using the `run_test.sh` script, which accepts a configuration name:
+
+```bash
+bash tests/e2e/run_test.sh gsm8k_function_rm
+```
+
+## GitHub Workflow
+
+The GitHub workflow has been reorganized into multiple jobs:
+
+1. **prepare_gsm8k**: Prepares the dataset and caches it for other jobs.
+2. **function_rm_tests**: Runs function-based reward model tests.
+3. **model_rm_tests**: Runs model-based reward model tests.
+4. **advanced_tests**: Runs specialized configurations like ReMax and Ulysses.
+
+## Adding New Test Configurations
+
+To add a new test configuration:
```

Comment:
we should add a section for "Contributing to verl" and include this doc as well as lint https://verl.readthedocs.io/en/latest/index.html#code-formatting in the website. Please include this doc in index.rst in the next PR